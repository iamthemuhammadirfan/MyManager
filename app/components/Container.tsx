import {SafeAreaView, StyleSheet, ViewStyle} from "react-native";
import React from "react";

type Props = {
  children: JSX.Element | JSX.Element[];
  style?: ViewStyle;
};

interface Styles {
  container: ViewStyle;
}

const Container: React.FC<Props> = ({children, style}: Props) => {
  return (
    <SafeAreaView style={[styles.container, style]} testID="container">
      {children}
    </SafeAreaView>
  );
};

export default Container;

const styles = StyleSheet.create<Styles>({
  container: {
    backgroundColor: "red",
    flex: 1,
  },
});
