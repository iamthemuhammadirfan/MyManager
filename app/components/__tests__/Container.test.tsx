import React from "react";
import {View} from "react-native";
import {render} from "@testing-library/react-native";

import Container from "components/Container";

describe("Container Component", () => {
  it("should match with the snap shoots", () => {
    const tree = render(
      <Container>
        <View />
      </Container>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should have a child component in it", () => {
    const {getByTestId} = render(
      <Container>
        <View testID="child-component" />
      </Container>,
    );

    const childComponent = getByTestId("child-component");

    expect(childComponent.props.children).toBe(
      Container.defaultProps?.children,
    );
  });
});
