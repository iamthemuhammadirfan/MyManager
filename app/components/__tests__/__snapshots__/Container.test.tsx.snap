// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should match with the snap shoots 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "backgroundColor": "red",
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="container"
>
  <View />
</RCTSafeAreaView>
`;
